OpenRA.Mods.Common/Activities/DeployForGrantedCondition.cs

```diff
@@ -19,48 +20,86 @@ public class DeployForGrantedCondition : Activity
 	{
 		readonly GrantConditionOnDeploy deploy;
 		readonly bool canTurn;
+		readonly Func<Activity> moveInner;
```

Comment:
Can you elaborate on the practical benefits of `Func<Activity>`s vs evaluating the activity up-front and either storing it here as an `Activity` or, I hope better, queuing it as `NextActivity`?

Reply:
The idea was for `Transforms` to evaluate the activity as late as possible to prevent trying to look up traits from a dead actor and causing crashes. This still didn't work out right though.

I like the idea of queuing the activity instead and have now used other means to prevent dead actor trait look-ups.